scripts/finetune.py

```diff
@@ -345,14 +365,12 @@ def terminate_handler(_, __, model):
     # TODO do we need this fix? https://huggingface.co/docs/accelerate/usage_guides/fsdp#saving-and-loading
     # only save on rank 0, otherwise it corrupts output on multi-GPU when multiple processes attempt to write the same file
     if cfg.fsdp:
-        model.save_pretrained(cfg.output_dir)
+        trainer.save_model(cfg.output_dir)
```

Comment:
this fixes an issue where the `.save_pretrained` call doesn't properly unwrap the model and leaves `_flat_param` modules in the saved model.